test/suites/ipv6/singlestack.robot

```diff
@@ -0,0 +1,102 @@
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${HOSTNAME}         hello-microshift.cluster.local
+
+
+*** Test Cases ***
+Verify Router Serves IPv6
+    [Documentation]    Verify router is capable of serving ipv6 traffic.
+    [Setup]    Run Keywords
+    ...    Create Hello MicroShift Pod
+    ...    Expose Hello MicroShift Service Via Route
+    ...    Restart Router
+
+    Must Be Ipv6    ${USHIFT_HOST}
+    Wait Until Keyword Succeeds    10x    6s
+    ...    Access Hello Microshift Success    ${HTTP_PORT}
+
+    DNS Entry For Route Should Resolve
+
+    [Teardown]    Run Keywords
+    ...    Delete Hello MicroShift Route
+    ...    Delete Hello MicroShift Pod And Service
+    ...    Wait For Service Deletion With Timeout
+
+Verify All Services Are Ipv6
+    [Documentation]    Check all services are running IPv6 only
+
+    All Services Are Ipv6
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Login MicroShift Host
+    Setup Suite With Namespace
+    Wait Until Greenboot Health Check Exited
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Delete Hello MicroShift Route
+    [Documentation]    Delete route for cleanup.
+    Oc Delete    route/hello-microshift -n ${NAMESPACE}
+
+Wait For Service Deletion With Timeout
+    [Documentation]    Polls for service and endpoint by "app=hello-microshift" label. Fails if timeout
+    ...    expires. This check is unique to this test suite because each test here reuses the same namespace. Since
+    ...    the tests reuse the service name, a small race window exists between the teardown of one test and the setup
+    ...    of the next. This produces flakey failures when the service or endpoint names collide.
+    Wait Until Keyword Succeeds    30s    1s
+    ...    Network APIs With Test Label Are Gone
+
+Expose Hello MicroShift Service Via Route
+    [Documentation]    Expose the "hello microshift" application through the Route
+    Oc Expose    pod hello-microshift -n ${NAMESPACE}
+    Oc Expose    svc hello-microshift --hostname hello-microshift.cluster.local -n ${NAMESPACE}
+
+Network APIs With Test Label Are Gone
+    [Documentation]    Check for service and endpoint by "app=hello-microshift" label. Succeeds if response matches
+    ...    "No resources found in <namespace> namespace." Fail if not.
+    ${match_string}=    Catenate    No resources found in    ${NAMESPACE}    namespace.
+    ${match_string}=    Remove String    ${match_string}    "
+    ${response}=    Run With Kubeconfig    oc get svc,ep -l app\=hello-microshift -n ${NAMESPACE}
+    Should Be Equal As Strings    ${match_string}    ${response}    strip_spaces=True
+
+DNS Entry For Route Should Resolve
+    [Documentation]    Resolve hello-microshift route via mDNS from the hypervisor/RF runner.
+    ...    Expects RF runner host has opened port 5353 for libvirt zone.
+
+    ${result}=    Run Process
+    ...    avahi-resolve-host-name ${HOSTNAME}
```

Comment:
I see it returned `hello-microshift.cluster.local	2001:db8:ca7:ff::1522`. Do think it's possible to have a race condition with another test that would also expose `hello-microshift.cluster.local`?

Should we check if the IP we got is indeed IPv6?

Reply:
Good points, did both.